cpu/riscv_common/irq_arch.c

```diff
@@ -134,6 +134,10 @@ static void handle_trap(uword_t mcause)
             write_csr(mepc, return_pc + 4);
             break;
         }
+        case CAUSE_FAULT_FETCH:
+        case CAUSE_FAULT_LOAD:
+        case CAUSE_FAULT_STORE:
+            core_panic(PANIC_MEM_MANAGE, "MEM MANAGE HANDLER");
```

Comment:
```suggestion
#ifdef MODULE_PERIPH_PMP
        case CAUSE_FAULT_FETCH:
            core_panic(PANIC_MEM_MANAGE, "MEM MANAGE HANDLER (fetch)");
        case CAUSE_FAULT_LOAD:
            core_panic(PANIC_MEM_MANAGE, "MEM MANAGE HANDLER (load)");
        case CAUSE_FAULT_STORE:
            core_panic(PANIC_MEM_MANAGE, "MEM MANAGE HANDLER (store)");
#endif
```
does it make sense to keep that information around?

Reply:
Good idea. WIll need to adjust the tests the. Do you insist in the `ifdef MODULE_PERIPH_PMP`? I could imagine RIOT should always be able to show the exact fault cause, independent of the feature you are using.

Reply:
Ah but I thought those faults can only occur when a PMP is present - otherwise the strings would just waste ROM without the possibility to ever be called.